src/applicationLayer.c

```diff
@@ -25,14 +26,101 @@ Result del(char* key){
     return result;
 }
 
+int beg(){
+  FILE *exclusiveFile;
+
+  exclusiveFile = fopen(".exclusive","r");
+
+  if(exclusiveFile != NULL){
+      fclose(exclusiveFile);
+      return 1002;
+  }
+
+  exclusiveFile = fopen(".exclusive","w");
+
+  fprintf(exclusiveFile, "%d", getpid());
+  fclose(exclusiveFile);
+
+  return 0;
+}
+
+int end (){
+  FILE *exclusiveFile;
+  exclusiveFile = fopen(".exclusive","r");
+
+  if(exclusiveFile == NULL){
+       return 1002;
+  }
+  fclose(exclusiveFile);
+
+  remove(".exclusive");
+  return 0;
+}
+
+int checkExclusive(){
+  int pid = getpid();
+  int exclusiveID;
+  FILE *exclusiveFile;
+  exclusiveFile = fopen(".exclusive", "r");
+
+  if(exclusiveFile == NULL){
+       return 0;
+  }
+
+  fscanf(exclusiveFile,"%i",&exclusiveID);
+
+  if (exclusiveID == pid){
+    return 0;
+  }
+
+  return 1;
+}
+
 Result executeCommand( Command command){
     Result result;
     char *formatedValue;
+
+    int hasPermission = checkExclusive();
+    if (hasPermission==1) {
+        result.error_code = 1;
+        result.value="Another user has exclusive write permissions";
+        result.malloced=1;
+        return result;
+    }
+
+    if(strcmp(command.order, "BEG")==0){
+      int exclusiveStatus = beg();
+      if(exclusiveStatus==0){

```

Comment:
The indentation is not the same as in other lines. This repeats in several lines.